app/models/atlas.rb

```diff
@@ -0,0 +1,32 @@
+class Atlas < ActiveRecord::Base
+  belongs_to :taxon
+  has_many :exploded_atlas_places
+  has_many :atlas_alterations
+  
+  def places
+    exploded_place_ids_to_include, exploded_place_ids_to_exclude = get_exploded_place_ids_to_include_and_exclude
+    Place.
+      where( "(admin_level = 0 OR places.id IN (?))", exploded_place_ids_to_include ).
+      where( "places.id NOT IN (?)", exploded_place_ids_to_exclude )
+  end
+  
+  def presence_places
+    exploded_place_ids_to_include, exploded_place_ids_to_exclude = get_exploded_place_ids_to_include_and_exclude
+    Place.joins( listed_taxa: :list ).
+      where( "lists.type = 'CheckList'").
+      where( "lists.id = places.check_list_id" ).
+      where( "listed_taxa.taxon_id = ?", taxon ).
+      where( "(admin_level = 0 OR places.id IN (?))", exploded_place_ids_to_include ).
+      where( "places.id NOT IN (?)", exploded_place_ids_to_exclude )
+  end
+  
+  def get_exploded_place_ids_to_include_and_exclude
+    exploded_place_ids_to_include = [0]
+    exploded_place_ids_to_exclude = [0]
```

Comment:
The Place query you proposed here: https://github.com/inaturalist/inaturalist/issues/1154#issuecomment-250304586 doesn't work if exploded_place_ids_to_include = [] & exploded_place_ids_to_exclude = [] (e.g. an atlas of only countries with nothing exploded) so adding a dummy place id of 0 (ie exploded_place_ids_to_include = [0] and exploded_place_ids_to_exclude = [0]) was a hack which fixes that case. But is a bit klutzy and things would break if there actually was a place with ID 0. Can you bear with the klutzyness? If not, can you alter your query in some other way that would work in the exploded_place_ids_to_include = [] & exploded_place_ids_to_exclude = []  case?